server/src-lib/Hasura/SQL/Value.hs

```diff
@@ -43,6 +43,7 @@ data PGColValue
   | PGValJSON !Q.JSON
   | PGValJSONB !Q.JSONB
   | PGValGeo !GeometryWithCRS
+  | PGValRaster !T.Text
```

Comment:
I think it would be nice to use a more specific type than `Text` here. Looking at the PostGIS documentation, the format of raster literals appears to use a hexadecimal representation of the [WellKnownBinary (WKB) format](https://github.com/gravitystorm/postgis/blob/master/raster/doc/RFC2-WellKnownBinaryFormat). While it probably doesn’t make sense to actually parse the binary format itself, maybe it would be nice to parse the hex string, since that’s easy to do. Using the [`text-conversions`](https://hackage.haskell.org/package/text-conversions-0.3.0/docs/Data-Text-Conversions.html) package (disclaimer: it’s my package!), we could define

```hs
newtype RasterWKB = RasterWKB { getRasterWKB :: Base16 ByteString }
```

and even derive `FromText` and `ToText` instances for free. That would give us at least slightly more type safety and possibly also slightly better error reporting if users supply invalid raster literals.

Reply:
Thank you. This helps me a lot. :slightly_smiling_face: